appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -135,6 +137,9 @@ internal class ExperienceRenderer(override val scope: AppcuesScope) : AppcuesCom
             // clear list in case this was a screen_view qualification
             potentiallyRenderableExperiences.clear()
             stateMachines.cleanup()
+
+            // and stop any Modal retry processing ongoing when screen changes
+            modalStateMachineOwner.stopRetryHandling()
```

Comment:
following on previous comment, we could remove this.

Reply:
maybe we can evolve this later, but the only thing I dont like about this entire appraoch is that the recoverable properties in Exception and StepError seems to be directly tied to basically waiting them to scroll to retry, so I'm trying to imagine if in the future we have other type of recoverable tasks (might never happen), how would the StateMachineOwning be able to differ one from the other, since they would both be StepError && recoverable

Reply:
yep, would certainly need to evolve if there are different types of recovery. Maybe some enum of recovery types, and this type is "UI changed" -- i.e. check again when you detect the UI changed in some meaningful way, rather than just one generic `recoverable` boolean.